dids/diddht/bencode/bencode.go

```diff
@@ -0,0 +1,260 @@
+package bencode
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+)
+
+const (
+	DictionaryPrefix = 'd'
+	IntegerPrefix    = 'i'
+	ListPrefix       = 'l'
+	EndSuffix        = 'e'
+)
+
+// Marshal encodes the given input into a Bencode formatted byte array.
+// More information about Bencode can be found at:
+// https://wiki.theory.org/BitTorrentSpecification#Bencoding
+func Marshal(input any) ([]byte, error) {
+	switch v := input.(type) {
+	case string:
+		encoded := fmt.Sprintf("%d:%s", len(v), v)
+
+		return []byte(encoded), nil
+	case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64:
+		encoded := fmt.Sprintf("%s%d%s", string(IntegerPrefix), v, string(EndSuffix))
+
+		return []byte(encoded), nil
+	case []byte:
+		size := fmt.Sprintf("%d:", len(v))
+		encoded := append([]byte(size), v...)
+
+		return encoded, nil
+	case []any:
+		var b []byte
+		b = append(b, ListPrefix)
+
+		for _, item := range v {
+			encoded, err := Marshal(item)
+			if err != nil {
+				return nil, err
+			}
+
+			b = append(b, encoded...)
```

Comment:
I'm not sure how efficient you want this to be, but if you pass a single`bytes.Buffer` around it will likely be significantly faster because you won't be allocating lots of tiny byte slices, instead you'll be writing to a single buffer.

If performance isn't a real concern than ignore.

Reply:
If you do do this you can then use `fmt.Fprintf` and fmt.Fscanf` directly on the buffer.

Reply:
great idea. performance isn't a big deal right now given that we're only ever marshaling a map with two fields that have primitive values. 

regardless, what you're suggesting is a better approach nonetheless. going to create an issue to track your suggestion so someone can tackle it